### PR TITLE
[FEATURE][native] Add badge count support for the linux platform

### PR DESCRIPTION
**Pull request by nirvn**
## Description
Unfortunately, no screenshot to offer, the badge doesn't show up on my gnome shell setup :)

@nyalldawson , thoughts?

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/native/linux/qgslinuxnative.cpp [hunk 1]
```diff
@@ -102,6 +102,22 @@ void QgsLinuxNative::hideApplicationProgress()
   QDBusConnection::sessionBus().send( message );
 }
 
+void QgsLinuxNative::setApplicationBadgeCount( int count )
+{
+  // the badge will only by shown when the count is greater than one
```
Comment (nyalldawson) [inline src/native/linux/qgslinuxnative.cpp]: I'd move this requirement up to app - regardless of the platform, we wouldn't want to show a badge count of 1

**Comment by nyalldawson** [comment]
Sorry - can't see any obvious reasons why this isn't working for you!

File: src/native/linux/qgslinuxnative.cpp (hunk 1 above)
Comment (nirvn) [reply src/native/linux/qgslinuxnative.cpp]: @nyalldawson , didn't we want to show a bade count of 1 on OS X since it doesn't have a progress indicator?

Comment (DelazJ) [reply src/native/linux/qgslinuxnative.cpp]: typo: s/by/be

**Comment by nirvn** [comment]
I can't get badges to show up on my system on any app. Closing this until the day I see badges on my monitor again :)

File: src/native/linux/qgslinuxnative.cpp [hunk 2]
```diff
@@ -102,6 +102,22 @@ void QgsLinuxNative::hideApplicationProgress()
...
+  const QVariantMap properties
+  {
+    { QStringLiteral( "count-visible" ), count > 1 },
+    { QStringLiteral( "count" ), count }
```
Comment (nyalldawson) [inline src/native/linux/qgslinuxnative.cpp]: static_cast< long long >( count )

works for me

**Comment by nyalldawson** [comment]
Proof:

peek 2018-09-06 11-04

**Comment by nirvn** [comment]
@nyalldawson , yep, casting does it. Commit updated. Should we merge when travis is green?

**Comment by nyalldawson** [comment]
One thing i'm not sure about is if we should hide the undefined progress state when the badge count is shown?

**Comment by nirvn** [comment]
@nyalldawson , hm, I'm not sure -- wouldn't it lead to this no so desirable sequence:
- at count 1, we'd have an undefined progress bar shown
- at count 2, we'd lose the progress bar and show a badge
- when it goes back to 1, we'd have an undefined progress bar shown again

?

**Comment by nyalldawson** [comment]
Maybe the logic should be that we never show the undefined progress bar, and always shows counts then? (even a count of 1)

**Comment by nirvn** [comment]
Maybe; I'd be -0 on that proposal, I like the undefined progress bar style on windows :)

**Comment by nyalldawson** [comment]
Me too, on Windows. On linux I just get a 0% progress bar (ugly!).

Anyway, I'm proposing that this logic sits in the native library itself, not in app. So it'd be linux specific.

**Comment by nirvn** [comment]
@3nids , the ball is in your court now :wink: it'd be great if OS X gets badge support in place of progress report.

**Pull request merged**